atlascope/core/job_types/gland_detection.py

```diff
@@ -0,0 +1,107 @@
+"""Detect likely locations of glands in the image."""
+import io
+
+from celery import shared_task
+from django.contrib.gis.geos import Point, Polygon
+import skimage.color
+import skimage.io
+import skimage.measure
+
+from atlascope.core.models import Dataset, DetectedStructure
+from atlascope.core.models.detected_structure import (
+    STRUCTURE_ATTRIBUTES,
+    structure_attribute_to_field_name,
+)
+
+from .nucleus_detection import detect_nuclei
+from .utils import save_output_dataset
+
+schema = {
+    "type": "object",
+    "required": [],
+    "properties": {
+        "gland_map_path": {
+            "type": 'string',
+            "title": 'Path to gland map',
+            "default": 'atlascope/core/management/populate/inputs/gland_map.jpg',
+        }
+    },
+}
+
+
+@shared_task
+def run(
+    job_id: str,
+    original_dataset_id: str,
+    gland_map_path=schema["properties"]["gland_map_path"]["default"],
+):
+    from atlascope.core.models import Job
+
+    original_dataset = Dataset.objects.get(id=original_dataset_id)
+    job = Job.objects.get(id=job_id)
+
+    def gland_data_subset(data: dict, subset_name: str):
+        return (
+            {
+                key.replace(subset_name, ''): value
+                for key, value in data.items()
+                if subset_name in key
+            },
+        )
+
+    try:
+        input_image = skimage.io.imread(io.BytesIO(open(gland_map_path, 'rb').read()))
+        glands, gland_mask = detect_nuclei(input_image)
+
+        detection_dataset = save_output_dataset(
+            original_dataset,
+            job.investigation,
+            'Detected Glands',
+            None,
+            {
+                'num_glands': len(glands),
+                'gland_mask': gland_mask.tolist(),
+            },
+            dataset_type='gland_detection',
+        )
+        for gland in glands:
+            additional_gland_attributes = {
+                structure_attribute_to_field_name(attribute): gland[attribute]
+                for attribute in STRUCTURE_ATTRIBUTES
+            }
+            # For small sample image, increase the coordinates of the
+            # centroids to better match the size of the image
+            centroid = Point(
+                x=gland['Identifier.CentroidX'] * 3,
+                y=gland['Identifier.CentroidY'] * 2.5,
+            )
```

Comment:
We sent the test image `management/populate/inputs/gland_map.png` into this detection. Our test image was much smaller than the associated dataset image, `HTA9_1_BA_F_ROI-2.ome.tiff` (About 3 times smaller horizontally and 2.5 times smaller vertically)  This simple translation of each centroid was merely to compensate for this size difference and move the detected glands to make them more spacially distributed within  `HTA9_1_BA_F_ROI-2.ome.tiff` (rather than isolated in the top right corner, where there are no nuclei).

Instead of further messing with this, I simply wrote a new `detect_glands` method, inspired by your previous comment. Instead of trying to reuse the nucleus detection on a test image, we can just generate where the glands go by generating n dots in a grid within the size of the input image. 

So this is fixed by [1827f2f](https://github.com/atlascope/atlascope/pull/193/commits/1827f2ffe5d832cd36859d7bdf4eff79ac1ae29c)